client/src/store/toolStore.js

```diff
@@ -21,34 +22,8 @@ const getters = {
     },
     getTools:
         (state) =>
-        ({ filterSettings }) => {
-            // if no filters
-            if (Object.keys(filterSettings).length == 0) {
-                return [];
-            }
-            const allTools = state.toolsList;
-            const returnedTools = [];
-
-            for (const tool in allTools) {
-                let hasMatches = false;
-                for (const [key, filterValue] of Object.entries(filterSettings)) {
-                    const actualValue = allTools[tool][key];
-                    if (filterValue) {
-                        if (!actualValue || !actualValue.toUpperCase().match(filterValue.toUpperCase())) {
-                            hasMatches = false;
-                            break;
-                        } else {
-                            hasMatches = true;
-                        }
-                    }
-                }
-                if (hasMatches) {
-                    returnedTools.push(allTools[tool]);
-                }
-            }
-
-            return returnedTools;
-        },
+        ({ toolbox }) =>
+            state.toolsList.length === 0 ? [] : filterTools(toolbox, state.toolsList),
```

Comment:
Should just return `filterTools`, the empty check should be in `filterTools`, and we could add a test for it if necessary.